### ISSUE TITLE
WARN: `failed to turn off translog retention` on startup

### ISSUE THREAD

**Issue by proddata** [crate/crate#12828]
### CrateDB version

5.0.0

### Steps to Reproduce

After upgrading a cluster from 4.7 to 5.0.0 we see lots of the following error messages on CrateDB startup

```
2022-07-26 17:18:09 | [2022-07-26T15:18:09,879][WARN ][o.e.i.s.IndexShard       ] [data-hot-0]  [schema.table][0]failed to turn off translog retention
-- | --
  |   | 2022-07-26 17:18:08 | at java.lang.Thread.run(Thread.java:833) [?:?]
  |   | 2022-07-26 17:18:08 | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
  |   | 2022-07-26 17:18:08 | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.index.shard.IndexShard$3.doRun(IndexShard.java:1861) [crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.index.shard.IndexShard.trimTranslog(IndexShard.java:1050) ~[crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.index.shard.IndexShard.getEngine(IndexShard.java:2453) ~[crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | org.apache.lucene.store.AlreadyClosedException: engine is closed
  |   | 2022-07-26 17:18:08 | [2022-07-26T15:18:08,977][WARN ][o.e.i.s.IndexShard       ] [data-hot-0]  [.partitioned.schema.table2][0]failed to turn off translog retention
  |   | 2022-07-26 17:18:08 | at java.lang.Thread.run(Thread.java:833) [?:?]
  |   | 2022-07-26 17:18:08 | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
  |   | 2022-07-26 17:18:08 | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.index.shard.IndexShard$3.doRun(IndexShard.java:1861) [crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.index.shard.IndexShard.trimTranslog(IndexShard.java:1050) ~[crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | at org.elasticsearch.index.shard.IndexShard.getEngine(IndexShard.java:2453) ~[crate-server.jar:?]
  |   | 2022-07-26 17:18:08 | org.apache.lucene.store.AlreadyClosedException: engine is closed
```

### Expected Result

no errors / warnings on startup after migrating from 4.x to 5.0

### Actual Result

```
9 | [2022-07-26T15:18:09,879][WARN ][o.e.i.s.IndexShard       ] [data-hot-0]  [schema.table][0]failed to turn off translog retention
-- | --

```

**Comment by matriv** [comment]
Those are `WARN` messages, not errors.

Did the ugprade run smoothly nevertherless?

**Comment by matriv** [comment]
If the update runs smoothly, then the issue is only the `WARN` logs which I wouldn't consider changing anything regarding those logs, as it could be helpful to troubleshoot real issues during upgrade.

If it causes much trouble to users, maybe we could define another logger for those, or mention them in an upgrade guide/how-to

**Comment by proddata** [comment]
I think it would make sense to maybe mention them.

**Comment by seut** [comment]
@proddata
The warning log message should not be written in some cases, fixed by #12982, but if so it can be ignored.
Thank you for reporting.
